File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteMiddle(ListNode head) {
        ListNode fast = head;
        ListNode slow = head;
        ListNode prev = null;

        if (fast == null) {return null;}
        if (fast.next == null) {return null;}
        
        while (fast != null) {
            fast = fast.next;
            if (fast == null) {
               break;
            }

            fast = fast.next;
            prev = slow;
            slow = slow.next;
        }

        if (prev!=null) {
            if (prev.next!= null){
                prev.next = prev.next.next;
            }else{
                prev.next = null;
            }
        }

        return head;
    }
}

File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteMiddle(head *ListNode) *ListNode {
    if head == nil || head.Next == nil {
        return nil
    }

    slow, fast := head, head
    var prev *ListNode

    for fast != nil {
        fast = fast.Next
        if fast != nil {
            fast = fast.Next
            prev = slow
            slow = slow.Next
        }
    }   

    if prev != nil && prev.Next != nil {
        prev.Next = prev.Next.Next
    }

    return head
}